docs/standards/javascript.md

```diff
@@ -0,0 +1,203 @@
+
+## Using CiviCRM Javascript in non CiviCRM pages
+
+If you are working outside the context of CiviCRM pages (e.g. on a Drupal page, Wordpress widget, Joomla page, etc) you need to explicitly tell CiviCRM to load its javascript in to the page header. You can add your own scripts as well.
+
+```php
+civicrm_initialize();
+$manager = CRM_Core_Resources::singleton();
+$manager->addCoreResources();
+$manager->addScriptUrl(WP_PLUGIN_URL . '/myplugin/myscript.js', 1, 'html-header');
+```
+
+Note when your using CiviCRM's resource manager to add scripts to non civicrm pages they need to be put as region 'html-header' because CiviCRM has no control over any of the other regions in non-civicrm pages.
+
+## Enclosing your code
+
+In Javacript all code needs to be closures to create the variable scope. Without closures, all variables would be global and there would be a significant risk of name collisions.
+
+The simplest closure you could write would be:
+
+```javascript
+CRM.$(function($) {
+  // your code here
+});
+```
+Remember that CRM.$ is our alias of jQuery. So the first line is shorthand notation for CRM.$('document').ready(function($) {
+The function receives jQuery as it's param, so now we have access to jQuery as the familiar $ for use in our code.
+
+If your code needs to work across multiple versions of CiviCRM, where jQuery was the older cj as well as the current CRM.$ you can use:
+
+```javascript
+(function($) {
+  // your code here
+})(CRM.$ || cj)
+```
+For more examples you can take a look at a [gist](https://gist.github.com/totten/9591b10d4bc09c78108d) from Tim Otten on javascript alternatives For more information on javascript closures, [here is some further reading](http://www.adequatelygood.com/2010/3/JavaScript-Module-Pattern-In-Depth).
+
+## Accessing CiviCRM API from JS
+
+If the current user has sufficient permissions (usually "Access CiviCRM") then your script can call the api and accomplish almost anything in CiviCRM. The syntax is:
+```javascript
+CRM.api('entity', 'action', {params}, {success: function}).
+```
+For more details, see [AJAX API](http://wiki.civicrm.org/confluence/display/CRMDOC43/AJAX+Interface) docs.
+
+## Server-Side Variables
+
+Just as you can dynamically add scripts to the page you can also add variables. They will be added to the CRM object in JSON format. Because this object is shared by many CiviCRM components, you should choose a unique namespace (typically the name of your extension). Example:
+
+```php
+// On the Server:
+CRM_Core_Resources::singleton()->addVars('myNamespace', array('foo' => 'bar'));
+```
+
+```javascript
+// In the JS file or inline script
+CRM.alert(CRM.vars.myNamespace.foo); // Alerts "bar"
+```
+
+## Localization
+
+As with PHP coding, any string that will be displayed to the user should be wrapped in ts() to translate the string. e.g. ts('hello')
```

Comment:
This section seems to have a lot of inline code-snippets that should marked with backquotes (eg `ts()` or `CRM_Core_Resources::addScriptFile()`).